Source/Client.swift

```diff
@@ -126,6 +128,9 @@ extension Client {
                 }
 
                 return response
+            } else if data.count == 0 {
+                //no data received, connection got broken
+                throw ClientError.brokenConnection
```

Comment:
Yeah exactly, if you manage to fix this up the stack that'd be even better, this *is* a bit of a hack, I just wanted to propose a quick fix before we find a proper solution.